tests/unit/core/test_withExperiment.js

```diff
@@ -0,0 +1,72 @@
+import { shallow } from 'enzyme';
+import * as React from 'react';
+
+import { withExperiment } from 'core/withExperiment';
+import { fakeCookie } from 'tests/unit/helpers';
+
+describe(__filename, () => {
+  describe('withExperiment', () => {
+    class SomeComponentBase extends React.Component {
+      render() {
+        return <div className="component" />;
+      }
+    }
+
+    function componentWithExperiment({ experimentProps, props } = {}) {
+      const SomeComponent = withExperiment({
+        id: 'ABtest',
+        variantA: 'AName',
+        variantB: 'BName',
+        ...experimentProps,
+      })(SomeComponentBase);
+
+      return shallow(<SomeComponent {...props} />);
+    }
+
+    it('passes the variant prop', () => {
+      const root = componentWithExperiment();
+      expect(root).toHaveProp('variant');
+    });
+
+    it('loads and saves cookie', () => {
```

Comment:
we should split this test case into multiple test cases:

- it loads a cookie upon construction
- it creates a cookie upon construction if none has been loaded
- it does not create a cookie upon construction if one has been loaded